tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -328,4 +364,75 @@ def montecarlo_main_loop(
         virt_packet_last_interaction_type,
         virt_packet_last_line_interaction_in_id,
         virt_packet_last_line_interaction_out_id,
+        rpacket_collections,
+    )
+
+
+def create_tracked_rpacket_df(rpacket_collections, iteration):
+    for i in range(len(rpacket_collections)):
+        if i == 0:
+            seed = rpacket_collections[0].seed
```

Comment:
Put this above the loop and iterate from 1 to len(rpacket_collections)

Reply:
I have made it such that it will explicitly create a array with the values from the `0` index and then append the other outputs to the previously created array. Thus, increasing the array dynamically.